docs/wiki/99 - Developers/Creating a Module.md

```diff
@@ -0,0 +1,147 @@
+# Creating a Module
+
+## Overview
+
+A Loris module consists of a group of pages that perform a specific function or assist with a specific task. Examples of modules include `Document Repository`, `Candidate List (Access Profiles)`, `Imaging Browser` ...
+
+
+## Directory structure
+
+Create a new folder corresponding to your module name under `modules/`. 
+This name should be short and descriptive. 
+The name should use only lowercase alphanumeric characters and underscores, e.g. `document_repository`.
+The directory name correponds to the URL that LORIS will use to load your module.
+For the remainder of this document we will assume your new module is named `my_new_module`.
+
+### PHP Files
+
+The following files are required for the module to load in LORIS:
+
+* A PHP file called "module": `php/module.class.inc`.
+* A PHP file with the same name as the module: `php/my_new_module.class.inc`.
+
+The namespace must be set to `LORIS\$modulename` for anything defined in the PHP directory. Filenames need to be lowercase in order to be autoloaded. Class names can be any case.
+
+The remaining PHP files will depend heavily on the kind of module you are creating. 
+
+The most common type of LORIS module will extend the class `NDB_Page` and belong to one of two types:
+"Menu Filters" and "Forms".
+
+#### Menu Filters
+
+This is a type of page that displays a table of data with options to filter the values of this table.
+
+The correct way to create a menu filter is to include additional classes that represent the idea of a
+row in the table displayed to a user. This differs from a row in the database itself; for example, a row in the 
+menu filter module may represent a joining of columns across many database tables.
+
+If you are creating a Menu Filter page, your class `php/my_new_module.class.inc` should extend
+the class `\DataFrameworkMenu`.
+
+[See this file for an example](../../../modules/dicom_archive/php/dicom_archive.class.inc).
+
+##### Data Instance
+You should create a class `php/mynewmodulerow.class.inc` to represent a Menu Filter row. This will
+provide basic structure for the row and include permission flags that can restrict data to users
+affiliated with a given Site or Project.
+
+
+[See this file for an example](../../../modules/dicom_archive/php/dicomarchiverow.class.inc).
+
+##### Row Provisioner.
+This file should be accompanied by a class `php/mynewmodulerowprovisioner.class.inc`. Provisioners usually
+contain the SQL statement(s) used to query the database for data that will be consolidated into a
+Menu Filter row.
+
+[See this file for an example](../../../modules/dicom_archive/php/dicomarchiverowprovisioner.class.inc).
```

Comment:
Note to self: I don't think it's possible to traverse up from this directory into the modules/ directory. 

Absolute links will probably be necessary.

Reply:
It is now.